src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,40 @@
-package core.mate.academy.service;
-
-/**
- * Your implementation of MachineService.
- */
-public class MachineServiceImpl {
-}
+package core.mate.academy.service;
+
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Your implementation of MachineService.
+ */
+public class MachineServiceImpl<T extends Machine> implements MachineService<T> {
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        if (type == Bulldozer.class) {

```

Comment:
instead of if-else
You can use multiple if statements, since you check exactly type.
And this make the code more readable.